framework/src/executioners/FixedPointSolve.C

```diff
@@ -0,0 +1,557 @@
+    _has_fixed_point_its(_max_fixed_point_its > 1),
+    _accept_max_it(getParam<bool>("accept_on_max_fixed_point_iteration")),
+    _has_fixed_point_norm(!getParam<bool>("disable_fixed_point_residual_norm_check")),
+    _fixed_point_rel_tol(getParam<Real>("fixed_point_rel_tol")),
+    _fixed_point_abs_tol(getParam<Real>("fixed_point_abs_tol")),
+    _fixed_point_force_norms(getParam<bool>("fixed_point_force_norms")),
+    _fixed_point_custom_pp(isParamValid("custom_pp") ? &getPostprocessorValue("custom_pp")
+                                                     : nullptr),
+    _relax_factor(getParam<Real>("relaxation_factor")),
+    _transformed_vars(getParam<std::vector<std::string>>("transformed_variables")),
+    _transformed_pps(getParam<std::vector<std::string>>("transformed_postprocessors")),
+    // this value will be set by MultiApp
+    _secondary_relaxation_factor(1.0),
+    _fixed_point_it(0),
+    _fixed_point_status(MooseFixedPointConvergenceReason::UNSOLVED),
+    _custom_rel_tol(getParam<Real>("custom_rel_tol")),
+    _custom_abs_tol(getParam<Real>("custom_abs_tol")),
+    _pp_old(0.0),
+    _pp_new(std::numeric_limits<Real>::max()),
+    _pp_scaling(1),
+    _max_xfem_update(getParam<unsigned int>("max_xfem_update")),
+    _update_xfem_at_timestep_begin(getParam<bool>("update_xfem_at_timestep_begin")),
+    _fixed_point_timer(registerTimedSection("FixedPointSolve", 1)),
+    _xfem_update_count(0),
+    _xfem_repeat_step(false),
+    _old_entering_time(_problem.time() - 1),
+    _solve_message(_problem.shouldSolve() ? "Solve Converged!" : "Solve Skipped!"),
+    _fail_step(false),
+    _auto_advance_set_by_user(isParamValid("auto_advance")),
+    _auto_advance_user_value(_auto_advance_set_by_user ? getParam<bool>("auto_advance") : true)
+{
+  // Handle deprecated parameters
+  if (!parameters().isParamSetByAddParam("relaxed_variables"))
+    _transformed_vars = getParam<std::vector<std::string>>("relaxed_variables");
+
+  if (_min_fixed_point_its > _max_fixed_point_its)
+    paramError("fixed_point_min_its",
+               "The minimum number of fixed point iterations may not exceed the maximum.");
+
+  if (_transformed_vars.size() > 0 && _transformed_pps.size() > 0)
+    mooseWarning(
+        "Both variable and postprocessor transformation are active. If the two share dofs, the "
+        "transformation will not be correct.");
+}
+
+bool
+FixedPointSolve::solve()
+{
+  TIME_SECTION(_fixed_point_timer);
+
+  Real current_dt = _problem.dt();
+
+  _fixed_point_timestep_begin_norm.clear();
+  _fixed_point_timestep_end_norm.clear();
+  _fixed_point_timestep_begin_norm.resize(_max_fixed_point_its);
+  _fixed_point_timestep_end_norm.resize(_max_fixed_point_its);
+
+  bool converged = true;
+
+  // need to back up multi-apps even when not doing fixed point iteration for recovering from failed
+  // multiapp solve
+  _problem.backupMultiApps(EXEC_TIMESTEP_BEGIN);
+  _problem.backupMultiApps(EXEC_TIMESTEP_END);
+
```

Comment:
Do we really need to a double backup here ?

Reply:
I did not change that part.